### PR TITLE
Update to Go 1.16 and consume index manager library

### PR DESCRIPTION
**Pull request by alexashley**
Removes the embedded index manager code in favor of the new library.  I ended up using some Go 1.16 specific types after pulling out the code, which forced the upgrade.

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
       - name: GoReleaser
         uses: goreleaser/goreleaser-action@v2
         with: 
-          version: v0.155.2
+          version: latest
```
Comment (alexashley) [inline .github/workflows/publish.yaml:29]: In #39, which was around the time of the Go 1.16 release, we had pinned the version of goreleaser as a workaround for an upstream issue build Apple ARM binaries.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -37,7 +37,7 @@ jobs:
         run : docker-compose down
   license-check:
     runs-on: ubuntu-latest
-    container: golang:1.15
+    container: golang:1.16
     steps:
     - uses: actions/checkout@v2
     - name: Download License Utility
```
Comment (mrparkers) [inline .github/workflows/test.yaml:43]: should we `go install` here since we're on 1.16 now?

Comment (mrparkers) [reply .github/workflows/test.yaml:43]: (referencing the line below the one I commented on)

Comment (alexashley) [reply .github/workflows/test.yaml:43]: Nice catch!

**Review by mrparkers — APPROVED** [review]

**Pull request merged**